controller/controller_tool/tool/src/py/Controller.py

```diff
@@ -128,67 +127,67 @@ def is_valid_dds_domain(self, dds_domain):
     def init_dds(self, dds_domain):
         """Initialize DDS entities."""
         factory = fastdds.DomainParticipantFactory.get_instance()
-        participant_qos = fastdds.DomainParticipantQos()
-        factory.get_default_participant_qos(participant_qos)
-        participant_qos.name('DDSRecorderController')
+        self.participant_qos = fastdds.DomainParticipantQos()
+        factory.get_default_participant_qos(self.participant_qos)
+        self.participant_qos.name('DDSRecorderController')
```

Comment:
NIT:
It is not pythonic to declare class attributes outside __init__ . 
Some linters would argue about this practice:
https://vald-phoenix.github.io/pylint-errors/plerr/errors/classes/W0201.html